libraries/node-core-library/src/Sort.ts

```diff
@@ -231,4 +263,100 @@ export class Sort {
       set.add(item);
     }
   }
+
+  /**
+   * Sort the keys given in an object
+   *
+   * @param object - The object to be sorted
+   * @param options - The options for sort keys
+   *
+   * @example
+   *
+   * ```ts
+   * console.log(Sort.sortKeys({ c: 3, b: 2, a: 1 })); // { a: 1, b: 2, c: 3}
+   * ```
+   */
+  public static sortKeys<T extends Partial<Record<string, unknown>> | unknown[]>(
+    object: T,
+    options: ISortKeysOptions = {
+      deep: false,
+      compare: Sort.compareByValue
+    }
+  ): T {
+    if (!isPlainObject(object) && !Array.isArray(object)) {
+      throw new TypeError(`Expected object or array`);
+    }
+    const cache: SortKeysCache = new WeakMap();
+    const context: ISortKeysContext = {
+      cache,
+      options
+    };
+
+    return Array.isArray(object)
+      ? (innerSortArray(object, context) as T)
+      : (innerSortKeys(object, context) as T);
+  }
+}
+function isPlainObject(obj: unknown): obj is object {
+  return obj !== null && typeof obj === 'object';
+}
+
+function innerSortArray(arr: unknown[], context: ISortKeysContext): unknown[] {
+  const resultFromCache: undefined | Partial<Record<string, unknown>> | unknown[] = context.cache.get(arr);
+  if (resultFromCache !== undefined) {
+    return resultFromCache as unknown[];
+  }
+  const result: unknown[] = [];
+  context.cache.set(arr, result);
+  if (context.options.deep) {
+    result.push(
+      ...arr.map((entry) => {
+        if (Array.isArray(entry)) {
+          return innerSortArray(entry, context);
+        } else if (isPlainObject(entry)) {
+          return innerSortKeys(entry, context);
+        }
+        return entry;
+      })
+    );
+  } else {
+    result.push(...arr);
```

Comment:
Not safe to pass `...arr` in a function argument position due to stack overflow if `arr` gets large. Use a loop over `arr` with per-element calls to `push` instead.